reana_cluster/utils.py

```diff
@@ -74,3 +74,36 @@ def _validate_cluster_spec(cluster_spec):
         raise e
 
     return True
+
+
+def build_component_url(host, ports, insecure=False):
+    """Build component URL, choosing the right URI scheme.
+
+    :param host: Host domain name or ip.
+    :type host: str
+    :param ports: List of possible ports where to contact the host.
+    :type ports: list of strings
+    :param insecure: Whether to use an insecure URI scheme or not.
+    :type insecure: bool
+    :returns: Full URL to the component.
+    :rtype: str
+    """
+    def _discover_https_port(ports):
+        https_ports = [443, 30443]
+        https_port_index = None
+        for https_port in https_ports:
+            try:
+                https_port_index = ports.index(https_port)
+            except ValueError:
+                continue
+        return ports[https_port_index] if https_port_index else None
+
+    https_port = _discover_https_port(ports)
+    if https_port and not insecure:
+        scheme = 'https'
+        port = https_port
+    else:
+        scheme = 'http'
+        port = ports[0]
+
+    return '{scheme}://{host}:{port}'.format(scheme=scheme, ip=ip, port=port)
```

Comment:
`undefined name 'ip'`. It should be `ip=host`, right?

Reply:
Yes, I messed in last refactor, thanks! (updated)

Reply:
```
$ reana-cluster init
$ kubectl get pods
NAME                                   READY   STATUS    RESTARTS   AGE
db-d7b444c58-c5kt2                     1/1     Running   0          10s
job-controller-f7c485484-6qhh9         1/1     Running   0          11s
message-broker-5f55bd9c95-dlxgs        1/1     Running   0          11s
server-57cd858bbb-hmswq                2/2     Running   0          11s
wdb-7467797784-5h97b                   1/1     Running   0          10s
workflow-controller-59ff5f9bdd-kzntf   2/2     Running   0          10s

$ reana-cluster env --include-admin-token
export REANA_SERVER_URL=https://192.168.64.2:30443
export REANA_ACCESS_TOKEN=<token>

$ reana-client ping
Could not connect to the selected REANA cluster server at https://192.168.64.2:30443
```

I can't make it to work. Am I missing something?